stl/inc/condition_variable

```diff
@@ -178,20 +155,10 @@ private:
 
     template <class _Mutex, class _Clock, class _Duration, class _Predicate>
     bool _Wait_until1(_Mutex& _Xtrnl, const chrono::time_point<_Clock, _Duration>& _Abs_time, _Predicate& _Pred) {
-        while (!_Pred()) {
-            const auto _Now = _Clock::now();
-            if (_Abs_time <= _Now) {
-                return false;
-            }
-
-            _CSTD xtime _Tgt;
-            const bool _Clamped = _To_xtime_10_day_clamped(_Tgt, _Abs_time - _Now);
-            if (wait_until(_Xtrnl, &_Tgt) == cv_status::timeout && !_Clamped) {
-                return _Pred();
-            }
-        }
-
-        return true;
+        const auto _Now = _Clock::now();
+        _CSTD xtime _Tgt;
+        (void) _To_xtime_10_day_clamped(_Tgt, _Abs_time - _Now);
+        return _Wait_until1(_Xtrnl, &_Tgt, _Pred);

```

Comment:
Ditto "This damages correctness if `_Abs_time` is more than 10 days in the future" by prematurely returning even if `_Pred` is `false`.

Reply:
Nope. There's more to fix: the Standard says this (well, the pretty function anyway) returns immediately when `_Pred()` is `true`, for example. I think:
```c++
while (!_Pred()) {
    const auto _Now = _Clock::now();
    _CSTD xtime _Tgt;
    const bool _Clamped = _To_xtime_10_day_clamped(_Tgt, _Abs_time - _Now);
    const cv_status _Result = _Wait_until1(_Xtrnl, &_Tgt, _Pred);
    if (!_Clamped && _Result == cv_status::timeout) {
        return _Pred();
    }
}
return true;
```